go/channelling/room_manager.go

```diff
@@ -209,9 +209,27 @@ func (rooms *roomManager) Get(roomID string) (room RoomWorker, ok bool) {
 	return
 }
 
+func (rooms *roomManager) isPublicRoom(roomName string) bool {
+	if rooms.PublicRoomNames != nil &&
+		rooms.PublicRoomNames.MatchString(roomName) {
+		return true
+	}
+	return false
+}
+
 func (rooms *roomManager) GetOrCreate(roomID, roomName, roomType string, credentials *DataRoomCredentials, sessionAuthenticated bool) (RoomWorker, error) {
+	isPublic := -1 // Will be evaluated lazily
```

Comment:
No, there are two cases where the value could be needed (1. when joining, 2. when creating a non-existing room). I only want to call `isPublicRoom` once, and only if necessary, i.e. "lazily".

So "not called yet" is flagged as value `-1`, "called and public" as value `1`and "called and private" as `0`.

Applies to here and all your comments below.